src/tests/cli-tests/tests/asm.test.ts

```diff
@@ -0,0 +1,41 @@
+import {executeCommand} from "../src/helper";
+import { paths } from '../src/entities';
+
+
+//id1746
+describe("Run with --asm by default", () => {
```

Comment:
Just to clarify, will each test be a piece of TypeScript code calling the binary?
Is it different from the solution we discussed initially? I've heard something about Allure that cannot be used for open-source repos.

Reply:
It is interesting that each test is a separate CI workflow.
Is it common practice?

Reply:
1. _will each test be a piece of TypeScript code calling the binary?_ - here the binary is just a stub for faster debugging. How I'm going to change the workflow: I would suggest to have 3 scenario:
 - 1. CLI tests will be called in any new PRs in case of anything in the repo was changed (**except changes in cli-tests**), **binaries will be build from sources**. Using this way we will have the latest build that was prepared by a PR and covered by tests
 - 2. CLI tests will be called in any new PRs in case of anything in the cli-tests dir was changed (**except whole project sources**), workflow will use latest prebuild binaries in this case.
 - 3. Manual run with any version of prebuild binaries
2. _Is it different from the solution we discussed initially?_ - based on description above I see no differences with initial discussion, is anything was missed? 
3. _I've heard something about Allure that cannot be used for open-source repos._ - We are in progress with this global question, with the new test case management system we will use another version and test cases will have public access.
4.  _It is interesting that each test is now a separate CI workflow. Is it a common practice?_ - In many cases it is a common practice, it helps to be transparent in results and easily handle tests. At the same time it is a flexible approach that can be joined or divided by situation/demand.

Reply:
@pcheremu understood, thanks!